website/content/roadmap.md

```diff
@@ -15,77 +15,133 @@ url: roadmap
-It contains only basic arithmetic functions (`AVERAGE`, `SUM`) and lacks
-functions for string manipulation, date handling and many math functions. One
-challenge here is the variety of functionalities and syntax of target DBMSs;
-e.g. there's no standard regex function. Improving our testing framework to
-include integration tests will help give us confidence here.
+The PRQL language can offer a vastly improved developer experience over SQL,
+both when exploring data and building robust data pipelines. We'd like to offer
+autocomplete both for PRQL itself and for columns of the underlying database,
+because fast iteration cycle can drastically decrease frustrations caused by
+banal misspellings.
 
-## Milestones
+This requires development across multiple dimensions — writing an
+[LSP server](https://langserver.org/), better support for typing in the
+compiler, and possibly database cohesion.
 
-We have assigned features to a broad timeline in our
-[Milestones.](https://github.com/PRQL/prql/milestones)
+While PRQL compiler will never depend on a database to compile queries, LPS
+server could greatly help with generating type definitions from the information
+schema of a database.
 
-<--->
+#### Query transparency
 
-## Alternative backends
+PRQL's compiler already contains structured data about the query. We'd like to
+offer transparency to tools which use PRQL, so they can offer lineage
+information, such as which tables are queried, and a DAG of transformations for
+each column.
 
-Currently, PRQL only transpiles into SQL. It could be much more powerful (and in
-some cases performant) if we develop a data-frame-handling-library backend. To
-be more precise, we would want to apply PRQL's AST to a in-memory dataframe of a
-performance-optimized library (such as [Polars](https://www.pola.rs/)).
+{{< /columns >}}
+
+## Long term
+
+{{< columns >}}
+
+#### SQL-to-PRQL conversion
+
+While PRQL already allows for a gradual on-ramp — there's no need to switch
+everything to PRQL right away — it would also be useful to be able to convert
+existing SQL queries to PRQL, rather than having to rewrite them manually. For
+many queries, this should be fairly easy. (For some it will be very difficult,
+but we can start with the easy ones...)
+
+#### Language
+
+While the core semantics and syntax of the language are now fairly stable, we
+are planning
+[a few major features](https://github.com/PRQL/prql/issues?q=is%3Aopen+is%3Aissue+label%3Amajor-feature+label%3Alanguage-design)
+that will give PRQL the feeling of a real programming language and elevate it in
+[the chomsky hierarchy](https://en.wikipedia.org/wiki/Chomsky_hierarchy).
+Honorable mentions here are recursive CTEs (or rather functions), algebraic type
+system, pre-specified join conditions and regex.
+
+Note that these features will probably inflict breaking changes with each minor
+release before we stabilize the 1.0, the first indefinitely supported language
+edition.
```

Comment:
@max-sixty I've removed "Rethinking joins". I feel that this is just one of the language features.

Reply:
OK, agree